### PR TITLE
Add missing configuration options to the box.cfg reference page

### LINKED ISSUE #4006: Document Enterprise box.cfg options and options introduced in 3.0
**Product:** Tarantool
**Since:** 3.0
**Root document:** https://www.tarantool.io/en/doc/latest/reference/configuration/
**SME:** @ totktonada

# Details

- Tarantool Enterprise options:
   - [x] Audit log (add after https://github.com/tarantool/doc/pull/3983)
   - [x] Authentication
   - [x] SSL options
   - [x] flight recorder
   - [x] WAL extensions
- Options introduced in 3.0, for example:
   - [x] secure_erasing
   - [x] auth_retries
   - [x] bootstrap_leader and bootstrap_strategy

### PR DESCRIPTION
**Pull request by andreyaksenov**
New sections (EE):
- [Authentication](https://docs.d.tarantool.io/en/doc/sync-box-cfg-params/reference/configuration/#authentication)
- [Flight recorder](https://docs.d.tarantool.io/en/doc/sync-box-cfg-params/reference/configuration/#flight-recorder)

New WAL options:
- [wal_ext](https://docs.d.tarantool.io/en/doc/sync-box-cfg-params/reference/configuration/#cfg-binary-logging-snapshots-wal-ext)
- [secure_erasing](https://docs.d.tarantool.io/en/doc/sync-box-cfg-params/reference/configuration/#cfg-binary-logging-secure-erasing)

Replication options:
- (updated) [bootstrap_strategy](https://docs.d.tarantool.io/en/doc/sync-box-cfg-params/reference/configuration/#cfg-replication-bootstrap-strategy)
- (new) [bootstrap_leader](https://docs.d.tarantool.io/en/doc/sync-box-cfg-params/reference/configuration/#cfg-replication-bootstrap-leader)

### CODE REVIEW

File: doc/reference/configuration/cfg_flight_recorder.rst [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+.. confval:: flightrec_logs_size
+
+    Since :doc:`2.11.0 </release/2.11.0>`.
+
+    Specifies the size (in bytes) of the log storage.
+    You can set this option to ``0`` to disable the log storage.
+
+    | Type: integer
+    | Default: 10485760
+    | Environment variable: TT_FLIGHTREC_LOGS_SIZE
```
Comment (xuniq) [inline doc/reference/configuration/cfg_flight_recorder.rst:44]: As I understand, all `flightrec` options are dynamic: https://github.com/tarantool/enterprise_doc/issues/178

```suggestion
    | Environment variable: TT_FLIGHTREC_LOGS_SIZE
    | Dynamic: **yes**
```

File: doc/reference/configuration/cfg_binary_logging_snapshots.rst [hunk 2]
```diff
@@ -136,4 +146,67 @@
...
+        so only new tuples are added to the log for ``space1`` and only old tuples for ``space2``.
+
+    Note that records with additional fields are :ref:`replicated <replication-architecture>` as follows:
+
+    *   If a replica doesn't support the extended format configured on a master, auxiliary fields are skipped.
+    *   If a replica and master have different configurations for WAL records, a master's configuration is ignored.
+
+    | Type: map
+    | Default: nil
+    | Environment variable: TT_WAL_EXT
```
Comment (xuniq) [inline doc/reference/configuration/cfg_binary_logging_snapshots.rst:197]: Is the option dynamic?

File: doc/reference/configuration/cfg_binary_logging_snapshots.rst [hunk 3]
```diff
@@ -136,4 +146,67 @@
...
+    | Dynamic: **yes**
+
+
+.. _cfg_binary_logging_snapshots-wal_ext:
+
+.. confval:: wal_ext
+
+    Since version :doc:`2.11.0 </release/2.11.0>`.
+
+    (Enterprise Edition only) Allows you to add auxiliary information to each :ref:`write-ahead log <internals-wal>` record.
```
Comment (xuniq) [inline doc/reference/configuration/cfg_binary_logging_snapshots.rst]:
```suggestion
    (**Enterprise Edition only**) Allows you to add auxiliary information to each :ref:`write-ahead log <internals-wal>` record.
```
Or maybe add a note - like in the [flightrec](https://github.com/tarantool/doc/pull/4007/files#diff-d5de9cc345b464252e9b7e67b6394f265b8bd6b183b08d98100a8667e606afc3R3) section ? Same for `secure_erasing`

File: doc/reference/configuration/cfg_binary_logging_snapshots.rst (hunk 2 above)
Comment (andreyaksenov) [reply doc/reference/configuration/cfg_binary_logging_snapshots.rst:197]: tarantool-ee tests say that yes, added

**Review by xuniq — APPROVED** [review]

**Pull request merged**